src/ArraySetops.chpl

```diff
@@ -0,0 +1,135 @@
+/* Array set operations
+ includes intersection, union, xor, and diff
+
+ currently, only performs operations with integer arrays 
+ */
+
+module ArraySetops
+{
+    use ServerConfig;
+
+    use SymArrayDmap;
+
+    use RadixSortLSD;
+    use Unique;
+    use Indexing;
+    use In1d;
+
+
+    /*
+    Small bound const. Brute force in1d implementation recommended.
+    */
+    private config const sBound = 2**4; 
+
+    /*
+    Medium bound const. Per locale associative domain in1d implementation recommended.
+    */
+    private config const mBound = 2**25; 
+
+    // returns intersection of 2 arrays
+    proc intersect1d(a: [] int, b: [] int, assume_unique: bool) {
+      //if not unique, unique sort arrays then perform operation
+      if (!assume_unique) {
+        var a1  = uniqueSort(a, false);
+        var b1  = uniqueSort(b, false);
+        return intersect1dHelper(a1, b1);
+      }
+      return intersect1dHelper(a,b);
+    }
+
+    // Get intersect of 2 arrays
+    // first concatenates the 2 arrays, then
+    // sorts arrays and removes all values that
+    // only occur once
+    proc intersect1dHelper(a: [] ?t, b: [] t) {
+      var aux = radixSortLSD_keys(concatset(a,b));
+
+      var head = sliceEnd(aux);
```

Comment:
maybe you should rename `sliceEnd()` to `sliceHead()`
and `tail()` to `sliceTail()`
it took me a minute to figure this out

Reply:
Thanks for the suggestion, I struggled with the naming of those, so glad to settle on something.